### PR TITLE
Refactor EOGPass to have a more readable and modular structure

### PR DESCRIPTION
**Pull request by konradweiss**
Closing #3 The class is restructured and the else-if construct is replaced by a handler structure similar to the Handler in the different Frontends. Although this does not decrease the size of the file, it makes it more readable and more searchable. Extracting more of the code into other classes was currently not done and can be done after identifying functional groups.

### CODE REVIEW

**Comment by oxisto** [comment]
You seem to suffer from the heisenbug test, please consider rebasing to the current master.

**Comment by konradweiss** [comment]
> You seem to suffer from the heisenbug test (fixed in #117), please consider rebasing to the current master.

Thank you, for helping me with my suffering from quantum coding paradoxe.

File: src/main/java/de/fraunhofer/aisec/cpg/passes/EvaluationOrderGraphPass.java [hunk 1]
```diff
@@ -67,12 +69,58 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(EvaluationOrderGraphPass.class);
 
+  protected final HashMap<Class<? extends Node>, CallableInterface<? extends Node>> map =
```
Comment (oxisto) [inline src/main/java/de/fraunhofer/aisec/cpg/passes/EvaluationOrderGraphPass.java:72]: Please use abstract types for fields, i.e. `Map`.

Comment (oxisto) [inline src/main/java/de/fraunhofer/aisec/cpg/passes/EvaluationOrderGraphPass.java:72]:
```suggestion
  protected final Map<Class<? extends Node>, CallableInterface<? extends Node>> map =
```

**Comment by oxisto** [comment]
Documentation could still be approved a little bit but I think it is already a lot better than it way. LGTM.

**Comment by Masrepus** [comment]
+1 from my side, I like this style that is similar to our frontend handlers better than the old one

**Pull request merged**